src/DotNetEnv/Parser.cs

```diff
@@ -15,10 +15,26 @@ private static bool IsComment(string line)
             return line.Trim().StartsWith("#");
         }
 
+        private static string GetQuotedValue(string input, string delimeter)
+        {
+            if (input.Contains(delimeter))
+            {
+                int quoteStart = input.IndexOf(delimeter);
+                int quoteEnd = input.LastIndexOf(delimeter)+1;
+                return input.Substring(quoteStart, quoteEnd);
+            }
+            return input;
+        }
         private static string RemoveInlineComment(string line)
         {
-            int pos = line.IndexOf('#');
-            return pos >= 0 ? line.Substring(0, pos) : line;
+            string value = GetQuotedValue(line, "\"");
+            value = GetQuotedValue(value, "\'");
```

Comment:
The face that you do both -- and do not check for them starting and ending the string before a comment -- means that now no one can include a single or double quoted string in an env var.  We need to handle cases like:

`"This isn't working"`
Or
`'Hi "Bob"!'`

And probably need to handle:
`"Hi \"Bob\"!"`
And
`'This isn\'t working'`